packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -172,6 +172,50 @@ const runAPI = (plugin, api, args, activity) => {
     }
     const localReporter = getLocalReporter(activity, reporter)
 
+    let runningActivities = new Set()
+
+    const localReporterThatCleansUpAfterMisbahavingPlugins = {
```

Comment:
this is yikes - I don't like recreating this in wrapping `activity.start/end` like that, but don't really see fundamentally better way to do this